### PR TITLE
[processes] Add mock for RemoteSysProbeUtil and tests for Process checks

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

Creates a mock for `RemoteSysProbeUtil` and then tests it's usage in `mergeProcWithSysprobeStats`. At the moment, `mergeProcWithSysprobeStats` is the only usage of the `RemoteSysProbeUtil` in the process and realtime process checks.

### Motivation

Adding test coverage for the sysprobe util

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by xornivore — APPROVED** [review]
Nice 🚀  thank you 👍

File: pkg/process/checks/remote_sysprobe_util_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+func TestMergeProcWithSysprobeStats(t *testing.T) {
+	assertMatchesSysProbeStats := func(t *testing.T, proc *procutil.Process, stats *process.ProcStatsWithPerm) {
+		assert.Equal(t, stats.OpenFDCount, proc.Stats.OpenFdCount)
+		assert.Equal(t, stats.ReadCount, proc.Stats.IOStat.ReadCount)
+		assert.Equal(t, stats.WriteCount, proc.Stats.IOStat.WriteCount)
+		assert.Equal(t, stats.ReadBytes, proc.Stats.IOStat.ReadBytes)
+		assert.Equal(t, stats.WriteBytes, proc.Stats.IOStat.WriteBytes)
+	}
+	hasSysProbeStats := func(proc *procutil.Process) bool {
+		return proc.Stats.IOStat.IsZeroValue() && proc.Stats.OpenFdCount != 0
```
Comment (robertjli) [inline pkg/process/checks/remote_sysprobe_util_test.go:39]:
```suggestion
		return !proc.Stats.IOStat.IsZeroValue() && proc.Stats.OpenFdCount != 0
```
We should be checking that it's _not_ the zero value, right?

File: pkg/process/checks/remote_sysprobe_util_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (robertjli) [inline pkg/process/checks/remote_sysprobe_util_test.go:1]: Optional nit: rename file to `sysprobe_stats_test.go`, as we don't use the RemoteSysProbeUtil.

File: pkg/process/checks/remote_sysprobe_util_test.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		proc2Stats := makeProcStatsWithPerm(2)
+
+		mockSysProbe := netMocks.NewSysProbeUtil(t)
+		mockSysProbe.On("GetProcStats", []int32{1, 2}).Return(&process.ProcStatsWithPermByPID{
+			StatsByPID: map[int32]*process.ProcStatsWithPerm{1: proc1Stats, 2: proc2Stats},
+		}, nil)
+
+		mergeProcWithSysprobeStats([]int32{1, 2}, map[int32]*procutil.Process{2: proc2}, mockSysProbe)
+
+		assert.False(t, hasSysProbeStats(proc2))
```
Comment (robertjli) [inline pkg/process/checks/remote_sysprobe_util_test.go:90]: Shouldn't this return stats for the proc that's requested?

**Review by robertjli — CHANGES_REQUESTED** [review]

File: pkg/process/checks/remote_sysprobe_util_test.go (hunk 1 above)
Comment (just-chillin) [reply pkg/process/checks/remote_sysprobe_util_test.go:39]: Ah yes, that's correct thank you

File: pkg/process/checks/remote_sysprobe_util_test.go (hunk 3 above)
Comment (just-chillin) [reply pkg/process/checks/remote_sysprobe_util_test.go]: Oh good catch, fixed!

**Review by robertjli — APPROVED** [review]

**Review by mbotarro — APPROVED** [review]

**Pull request merged**